modules/hikari/src/main/scala/doobie/hikari/Config.scala

```diff
@@ -103,4 +138,92 @@ object Config {
       c
     }
 
+  def apply(
+    jdbcUrl: String,
+    catalog: Option[String] = None,
+    connectionTimeout: Duration = Duration(30, TimeUnit.SECONDS),
+    idleTimeout: Duration = Duration(10, TimeUnit.MINUTES),
+    leakDetectionThreshold: Duration = Duration.Zero,
+    maximumPoolSize: Int = 10,
+    maxLifetime: Duration = Duration(30, TimeUnit.MINUTES),
+    minimumIdle: Int = 10,
+    password: Option[String] = None,
+    poolName: Option[String] = None,
+    username: Option[String] = None,
+    validationTimeout: Duration = Duration(5, TimeUnit.SECONDS),
+    allowPoolSuspension: Boolean = false,
+    autoCommit: Boolean = true,
+    connectionInitSql: Option[String] = None,
+    connectionTestQuery: Option[String] = None,
+    dataSourceClassName: Option[String] = None,
+    dataSourceJNDI: Option[String] = None,
+    driverClassName: Option[String] = None,
+    initializationFailTimeout: Duration = Duration(1, TimeUnit.MILLISECONDS),
+    isolateInternalQueries: Boolean = false,
+    readOnly: Boolean = false,
+    registerMbeans: Boolean = false,
+    schema: Option[String] = None,
+    transactionIsolation: Option[TransactionIsolation] = None,
+  ): Config = new Config(
+    jdbcUrl,
+    catalog,
+    connectionTimeout,
+    idleTimeout,
+    leakDetectionThreshold,
+    maximumPoolSize,
+    maxLifetime,
+    minimumIdle,
+    password,
+    poolName,
+    username,
+    validationTimeout,
+    allowPoolSuspension,
+    autoCommit,
+    connectionInitSql,
+    connectionTestQuery,
+    dataSourceClassName,
+    dataSourceJNDI,
+    driverClassName,
+    initializationFailTimeout,
+    isolateInternalQueries,
+    readOnly,
+    registerMbeans,
+    schema,
+    transactionIsolation
+  )
+
+  @nowarn
+  def fromProduct(p: Product): Config = p.productArity match {
```

Comment:
Sorry not sure why we need this?

Reply:
Scala 3 defines `fromProduct` in companion object of case classes. If you add a new field to the case class, you need to add a new case to `fromProduct` to ensure compatibility with old code which counts with the old number of fields.
I've improved the comment on top of the case class.